### DISCUSSION TITLE
Thoughts on using Typescript and SvelteKit's `package` command?

### DISCUSSION THREAD

**Discussion by jrmoynihan** [Cweili/svelte-fa#250] [Ideas]
Hey there @Cweili , I've noticed that a fair number of the issues that arise in this package are the result of imported/exported members and their type definitions, especially when using SvelteKit. Presumably, this is related to the ESM-centric model that Vite uses.  I don't think the `svelte-fa` package exports all of its components _as modules_ in the package.json.

I was wondering if you've considered adding Typescript directly to the component's script tags (not just the type definitions) and using SvelteKit's built-in [`package`](https://kit.svelte.dev/docs/packaging) command to help reliably generate & manage the exported types and component exports for you.  It automatically generates the type definition files for your components upon build, and importantly **it adds the entry points of the components in the `package.json`** (under the `"exports"` field).  I think this is related to some of the auto-import & namespace issues people encounter when building with `svelte-fa` in Kit.  It also defaults the `"svelte"` field in `package.json` to `'./index.js'` rather than `'src/fa.svelte'`, which would probably solve the default import case as intended.  Likewise, the generated `package.json` uses `"type":"module"` to indicate that all exports are considered ES modules, which is what Vite/Kit expects.

It seems like the simple solution might just be to use this package-building tool the Svelte team designed to streamline the process for you.

I've already tried migrating over to the Kit `package` method in [my fork](https://github.com/jrmoynihan/svelte-fa), if you want to take a look.  I also removed what appeared to be redundant tools/files related to the build process (e.g. Rollup config is built-in as a plugin to the Kit build step), and added a few types that were not emitted/exported (e.g. FlipDirection and PullDirection).  It probably needs to have the tests added back in, however.

Let me know what you think?

**Comment by Cweili** [comment]
I'm sorry for just seeing this discussion now. The v4 version has already been built with SvelteKit. Thank you!

**Reply by jrmoynihan** [reply]
No worries!  Thanks for your efforts on this package!
